src/main/scala/pirate.internal/ParseTraversal.scala

```diff
@@ -176,17 +175,30 @@ object ParseTraversal {
         })
     }
 
-  def eval[A](p: Parse[A]): Option[A] = p match {
+  def parseError[A](arg: String): P[A] =
+    arg.toList match {
+      case '-' :: _ => errorMessageP(s"Invalid option `${arg}`")
+      case _        => errorMessageP(s"Invalid argument `${arg}`")
+    }
+
+  def eval[A](p: Parse[A]): ValidationNel[Parser[Unit], A] = p match {
```

Comment:
Does this really accumulate properly? If it does, at least needs a test. At least a bit skeptical about the behaviour of this around alt and whether it really works as expected when multiple branches are taken, which branch does it gather errors from?